test/index.js

```diff
@@ -766,6 +766,136 @@ describe('def', function() {
+                   'new Number(-9007199254740992)'));
+
+    eq(inc(1), 2);
+  });
+
+  it('supports the "NonZeroInteger" type', function() {
+    var def = $.create($.env.concat([$.ValidNumber, $.NonZeroInteger]));
+
+    //  div :: ValidNumber -> NonZeroInteger -> ValidNumber
+    var div = def('div',
+                  {},
+                  [$.ValidNumber, $.NonZeroInteger, $.ValidNumber],
+                  function(a, b) { return a / b; });
+
+    throws(function() { div(3, 0); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received 0'));
+
+    throws(function() { div(3, new Number(0)); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received new Number(0)'));
+
+    throws(function() { div(3, -0); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received -0'));
+
+    throws(function() { div(3, new Number(-0)); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroInteger ' +
+                   'as its second argument; received new Number(-0)'));
+
+    eq(div(4, 2), 2);
+  });
+
+  it('supports the "PositiveInteger" type', function() {
+    var def = $.create($.env.concat([$.PositiveInteger]));
+
+    //  inc :: PositiveInteger -> PositiveInteger
+    var inc = def('inc',
+                  {},
+                  [$.PositiveInteger, $.PositiveInteger],
+                  function(x) { return x + 1; });
+
+    throws(function() { inc(-1.5); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type PositiveInteger ' +
+                   'as its first argument; received -1.5'));
+
+    throws(function() { inc(new Number(-1.5)); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type PositiveInteger ' +
+                   'as its first argument; received new Number(-1.5)'));
+
+    throws(function() { inc(-1); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type PositiveInteger ' +
+                   'as its first argument; received -1'));
+
+    throws(function() { inc(new Number(-1)); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type PositiveInteger ' +
+                   'as its first argument; received new Number(-1)'));
+
+    eq(inc(1), 2);
+  });
+
+  it('supports the "NegativeInteger" type', function() {
+    var def = $.create($.env.concat([$.NegativeInteger]));
+
+    //  dec :: NegativeInteger -> NegativeInteger
+    var dec = def('dec',
+                  {},
+                  [$.NegativeInteger, $.NegativeInteger],
+                  function(x) { return x - 1; });
+
+    throws(function() { dec(1.5); },
```

Comment:
Same as above. We should use `-1.5` here instead to ensure negative non-integer values are rejected.